src/components/Web3Provider/index.tsx

```diff
@@ -40,6 +48,40 @@ function Tracer() {
     }
   }, [networkProvider, provider, shouldTrace])
 
+  // Send analytics events when the active account changes.
+  const previousAccount = usePrevious(account)
+  const getConnection = useGetConnection()
+  const [connectedWallets, addConnectedWallet] = useConnectedWallets()
+  useEffect(() => {
+    if (account && account !== previousAccount) {
+      const walletType = getConnection(connector).getName()
+      const peerWalletAgent = provider ? getWalletMeta(provider)?.agent : undefined
+      const isReconnect = connectedWallets.some(
+        (wallet) => wallet.account === account && wallet.walletType === walletType
+      )
+
+      // User properties *must* be set before sending corresponding event properties,
+      // so that the event contains the correct and up-to-date user properties.
+      user.set(CustomUserProperties.WALLET_ADDRESS, account)
+      user.set(CustomUserProperties.WALLET_TYPE, walletType)
+      user.set(CustomUserProperties.PEER_WALLET_AGENT, peerWalletAgent ?? '')
+      if (chainId) {
+        user.postInsert(CustomUserProperties.ALL_WALLET_CHAIN_IDS, chainId)
+      }
+      user.postInsert(CustomUserProperties.ALL_WALLET_ADDRESSES_CONNECTED, account)
+
+      sendAnalyticsEvent(InterfaceEventName.WALLET_CONNECT_TXN_COMPLETED, {
```

Comment:
This implies that a WalletConnect "connecting" tx has completed. It's been taken out of the context of the other WC events, so it makes less sense here, but I don't think we should change it in this PR because this is meant only to be a fix for previously broken behavior.